openwillis/measures/text/speech_attribute.py

```diff
@@ -104,58 +102,35 @@ def filter_transcribe(json_conf, measures, speaker_label=None):
         only the relevant data for processing.
     text_list: list
         List of transcribed text.
-         split into words, phrases, turns, and full text.
+         split into words, turns, and full text.
     text_indices: list
         List of indices for text_list.
-         for phrases and turns.
-    Raises:
-    ...........
-    ValueError: If the speaker label is not found in the json response object.
     ------------------------------------------------------------------------------------------------------
     """
     item_data = json_conf["results"]["items"]
-
-    # make a dictionary to map old indices to new indices
-    item_data = cutil.create_index_column(item_data, measures)
+    
+    for i, item in enumerate(item_data): # create_index_column
+        item[measures["old_index"]] = i
 
     # extract text
-    text = " ".join(
-        [
-            item["alternatives"][0]["content"]
-            for item in item_data
-            if "alternatives" in item
-        ]
-    )
-
-    # phrase-split
+    text = " ".join([item["alternatives"][0]["content"] for item in item_data if "alternatives" in item])
     phrases, phrases_idxs = cutil.phrase_split(text)
 
-    # turn-split
-    turns = []
-    turns_idxs = []
-
     if speaker_label is not None:
+        turns_idxs, turns = cutil.filter_speaker_aws(item_data, speaker_label)
+    else:
+        turns_idxs, turns = [], []
```

Comment:
```suggestion
        turns_idxs, turns = [], []
        text = " ".join(phrases)
```

Join phrases for text if only one speaker

Reply:
this is incorrect, we just need to replace phrases with turns